File: rater/objects/RatingsOptions.m
classdef RatingsOptions
  properties
    k
    c
    fTree
    fNames
    numFs
    tolRel
    tolScale
  end
  
  methods
    function rOptions = RatingsOptions(fTree, tolRel, tolScale)
      rOptions.tolRel = tolRel;
      rOptions.tolScale = tolScale;
      rOptions.fTree = fTree;
      rOptions.fNames = keys(rOptions.fTree);
      rOptions.numFs = length(rOptions.fNames);
    end
    
    function rOptions = update(rOptions, k, c, strFs)
      rOptions.k = k;
      rOptions.c = c;
      i = 1;
      
      while (i <= rOptions.numFs)
        strF = strFs(i);
        rOptions.fTree(rOptions.fNames{i}) = [strF 1 / strF];
        i = i + 1;
      end
    end
  end
end

File: rater/objects/RatingsOptions.R
newRatingsOptions <- function(fTree) {
  fNames <- keys(fTree)
  numFs <- length(fNames)

  rOptions <- list(
    ks=c(1.5, 1.5),
    c=0.2,
    model=newModel(),
    xpDefault=1,
    fTree=fTree,
    fNames=fNames,
    numFs=numFs,
    tolRel=0.01,
    tolScale=0.01
  )
  
  class(rOptions) <- "RatingsOptions"
  rOptions
}

updateOptions <- function(rOptions, ks, c, strBetas, strFsNorm) {
  rOptions$ks <- ks
  rOptions$c <- c
  rOptions$model <- updateModel(rOptions$model, strBetas)
  strFs <- exp(strFsNorm)
  i <- 1
  
  while (i <= rOptions$numFs) {
    strF <- strFs[i]
    rOptions$fTree[[rOptions$fNames[i]]] <- c(strF, 1 / strF)
    i <- i + 1
  }

  rOptions
}
